ctapipe/io/dl1eventsource.py

```diff
@@ -0,0 +1,389 @@
+from ctapipe.io.datalevels import DataLevel
+from ctapipe.instrument import (
+    TelescopeDescription,
+    SubarrayDescription,
+    CameraGeometry,
+    OpticsDescription,
+)
+from ctapipe.containers import EventAndMonDataContainer, EventType
+from ctapipe.containers import MCHeaderContainer, MCEventContainer
+from ctapipe.io.eventsource import EventSource
+from astropy.coordinates import Angle
+import astropy.units as u
+import tables
+import warnings
+
+
+class DL1EventSource(EventSource):
+    def __init__(
+        self,
+        input_url,
+        config=None,
+        parent=None,
+        **kwargs
+    ):
+        """
+        EventSource for dl1 files in the standard DL1 data format
+
+        Parameters:
+        -----------
+        input_url : str
+            Path of the file to load
+        config: ??
+        parent: ??
+        kwargs
+        """
+        super().__init__(
+            input_url=input_url,
+            config=config,
+            parent=parent,
+            **kwargs
+        )
+
+        self.file_ = tables.open_file(input_url)
+        self._subarray_info = self._prepare_subarray_info(
+            self.file_.root.configuration.instrument
+        )
+        self._mc_header = self._parse_mc_header(
+            self.file_.root.configuration.simulation.run
+        )
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.close()
+
+    def close(self):
+        self.file_.close()
+
+    @staticmethod
+    def is_compatible(file_path):
+        """
+        Implementation needed!
+        """
+        return False
+
+    @property
+    def is_simulation(self):
+        """
+        Implementation needed!
+        """
+        return True
+
+    @property
+    def subarray(self):
+        return self._subarray_info
+
+    @property
+    def datalevels(self):
+        return (DataLevel.DL1_IMAGES, DataLevel.DL1_PARAMETERS)
+
+    @property
+    def obs_id(self):
+        return set(self.file_.root.dl1.event.subarray.trigger.col("obs_id"))
+
+    @property
+    def mc_header(self):
+        return self._mc_header
+
+    def _generator(self):
+        try:
+            yield from self._generate_events()
+        except EOFError:
+            msg = 'EOFError reading from "{input_url}". Might be truncated'.format(
+                input_url=self.input_url
+            )
+            self.log.warning(msg)
+            warnings.warn(msg)
+
+    def _prepare_subarray_info(self, instrument_description):
+        """
+        Constructs a SubArrayDescription object from
+        self.file_.root.configuration.instrument.telescope.optics
+        and
+        self.file_.root.configuration.instrument.subarray.layout
+        tables.
+
+        Returns
+        -------
+        SubarrayDescription :
+            instrumental information
+        """
+        available_optics = instrument_description.telescope.optics.iterrows()
```

Comment:
You can make this whole function much simpler if you use astropy table, which is how it was intended to be used.  ll Instrument classes have a `from_table()` method to support creating a full SubarrayDescription without having to parse anything. 

For example: 

```py3
cam_table = Table.read(file, "/instrument/telescope/camera/LSTCam")
geom = CameraGeometry.from_table(cam_table)
```
In that way, all the headers are also read correctly, and units applied, etc.   I had started making a `SubarrayDescription.from_hdf5()` method to do all that, but got sidetracked at some point. I can check if I still have the code somewhere, or I can prepare something for you.

Reply:
It means mixing a bit pytables and astropy.table, but that's also how we wrote it (and astropy.table adds a special metadata dataset that is handled automatically by this function, and includes astropy units, in a JSON blob)

Reply:
Sounds good!
I'll try it out.

Reply:
Ill have a look into that.
Right now it seems that the missing `from_table` methods are in the way, so I will do that first

Reply:
Sounds good!

Since the OpticsDescription has no `from_table` method yet, I am looking into that right now. The CameraGeometries each have their own table, whereas the OpticDescriptions are all stored in one table. So one would need to implement a `from_row` function instead.
Am I missing something?

Reply:
Yes, I guess it would work pretty much like the OpticsDescription.from_name() function. In fact we could just modify that to  allow one to pass in a table - right now it reads from the default table in ctapipe-extra as a string, but could also accept a table directly. 

Something like:
```py3
optics_table=Table.read(outfile, "/instrument/telescope/optics")

od = OpticsDescription.from_name(
    name="SST-GCT", 
    optics_table=optics_table)
)
```

Reply:
Thats even better!

Reply:
see #1358